modules/secure-cd/main.tf

```diff
@@ -33,38 +33,138 @@ locals {
       for env in var.deploy_branch_clusters : env if env.project_id == project_id
     ]]
   )
+
+  cd_sa_required_roles = [
+    "roles/clouddeploy.jobRunner",
+  ]
+}
+
+# Cluod Deploy Execution Service Account
+# https://cloud.google.com/deploy/docs/cloud-deploy-service-account#execution_service_account
+resource "google_service_account" "clouddeploy_execution_sa" {
```

Comment:
nit: consider reorganizing/splitting the main.tf into a few tf file - maybe deploy.tf, build.tf etc with associated resources for readability